spotbugs/src/main/java/edu/umd/cs/findbugs/TextUICommandLine.java

```diff
@@ -343,35 +357,41 @@ protected void handleOption(String option, String optionExtraPart) {
             mergeSimilarWarnings = false;
         } else if ("-sortByClass".equals(option)) {
             bugReporterType = SORTING_REPORTER;
-            SortingBugReporter sortingBugReporter = new SortingBugReporter();
-            handleOutputFilePath(sortingBugReporter, optionExtraPart);
-            reporters.add(sortingBugReporter);
+            TextUIBugReporter reporter = initializeReporter(optionExtraPart,
+                    () -> new SortingBugReporter(),
+                    (r, rOpts) -> {
+                    });
+            if (reporter != null)
+                reporters.add(reporter);
         } else if ("-xml".equals(option)) {
             bugReporterType = XML_REPORTER;
-            XMLBugReporter xmlBugReporter = new XMLBugReporter(project);
-            optionExtraPart = handleOutputFilePath(xmlBugReporter, optionExtraPart);
-            boolean xmlWithMessages = false;
-            boolean xmlMinimal = false;
-            if (!"".equals(optionExtraPart)) {
-                if ("withMessages".equals(optionExtraPart)) {
-                    xmlWithMessages = true;
-                } else if ("withAbridgedMessages".equals(optionExtraPart)) {
-                    xmlWithMessages = true;
-                    xmlWithAbridgedMessages = true;
-                } else if ("minimal".equals(optionExtraPart)) {
-                    xmlMinimal = true;
-                } else {
-                    throw new IllegalArgumentException("Unknown option: -xml:" + optionExtraPart);
-                }
-            }
-            xmlBugReporter.setAddMessages(xmlWithMessages);
-            xmlBugReporter.setMinimalXML(xmlMinimal);
-            reporters.add(xmlBugReporter);
+            TextUIBugReporter reporter = initializeReporter(
+                    optionExtraPart,
+                    () -> new XMLBugReporter(project),
+                    (r, arg) -> {
+                        if (!"".equals(arg)) {
+                            if ("withMessages".equals(arg)) {
+                                r.setAddMessages(true);
+                            } else if ("withAbridgedMessages".equals(arg)) {
+                                r.setAddMessages(true);
+                                xmlWithAbridgedMessages = true;
+                            } else if ("minimal".equals(arg)) {
+                                r.setMinimalXML(true);
+                            } else {
+                                throw new IllegalArgumentException("Unknown option: -xml:" + arg);
+                            }
+                        }
+                    });
+            if (reporter != null)
```

Comment:
Anytime we have an if statement, just because there is only one item, please always add {} as expected by general google styling rules.   Like my first comment this makes it easier to read in the future.

Reply:
Updated